core/src/test/scala/wartremover/warts/NoNeedForMonadTest.scala

```diff
@@ -34,6 +34,14 @@ class NoNeedForMonadTest extends FunSuite {
       Option(3).flatMap { case t => Some(t) }
     }
 
+    val extendsFunction = WartTestTraverser(NoNeedForMonad) {
+      object test extends Function1[Int, Option[Int]] {
```

Comment:
Can we also add a `test` object that does not extend `Function1` but has method `apply`? Like this:

```scala
object test2 {
  def apply(i: Int) = Option(i + 4)
}

Option(2) flatMap test2
```